### ISSUE TITLE
Get rid of inline styles.

### ISSUE THREAD

**Issue by maximLyakhov** [dj-fiorex/angular2-smart-table#98]
Current implementation writes conditional styles directly, with _!important_ is only way to override them.
As _!important_ flag is a bad practice, it would be a lot easier to style table with at least conditional classes and having an option to override default table stylesheet.

**Comment by uap-universe** [comment]
I agree completely. That's something that bugged me in my use cases for the smart table as well.

I carefully mark this as a breaking change, though, because there might be users who already depend on the way was implemented by the original authors. But I can imagine that there is actually a compatible way to move the inline styles to proper css classes without breaking anything - but that's not so easy to verify imho. But if we can do that, the label can of course be removed.
